pkg/kubelet/metrics/collectors/volume_stats.go

```diff
@@ -120,7 +126,16 @@ func (collector *volumeStatsCollector) CollectWithStability(ch chan<- metrics.Me
 			addGauge(volumeStatsInodesDesc, pvcRef, float64(*volumeStat.Inodes))
 			addGauge(volumeStatsInodesFreeDesc, pvcRef, float64(*volumeStat.InodesFree))
 			addGauge(volumeStatsInodesUsedDesc, pvcRef, float64(*volumeStat.InodesUsed))
+			addGauge(volumeStatsHealthAbnormalDesc, pvcRef, convertBoolToFloat64(volumeStat.VolumeHealthStats.Abnormal))
```

Comment:
This generates
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1d5d77b]
goroutine 2564 [running]:
k8s.io/kubernetes/pkg/kubelet/metrics/collectors.(*volumeStatsCollector).CollectWithStability(0x0?, 0xc0016df860)
        pkg/kubelet/metrics/collectors/volume_stats.go:129 +0x89b
```

https://github.com/kubernetes/kubernetes/issues/108715#issuecomment-1069085550

Reply:
Well that makes sense since `volumeStat.VolumeHealthStats` looks like an optional field:

```
    // +optional
	VolumeHealthStats *VolumeHealthStats `json:"volumeHealthStats,omitempty"`
```

Reply:
@fengzixu Can you take a look at this?

Reply:
Sure. Let me check

Reply:
File this PR to fix: https://github.com/kubernetes/kubernetes/pull/108758